docs/getting-started/throwing-errors.md

```diff
@@ -0,0 +1,289 @@
+        if (!response.ok) {
+            throw new OperationalError({
+                code: 'POKEMON_API_FAILED',
+                message: `The Pokemon API responded with a ${response.status} status`,
+                relatesToSystems: ['pokeapi']
+            });
+        }
+
+        response.send(await response.json());
+
+    } catch (error) {
+        // Pass any errors onto the Express error handler
+        next(error);
+    }
+});
+```
+
+### Operational errors in library code
+
+Operational errors should not be used in places where an error may be caused by a programmer who's _using_ your library. For example, if you publish the following library to be used by other engineers:
+
+```js
+function uppercaseAll(array) {
+    return array.map(item => item.toUpperCase());
+}
+```
+
+Then you may want to add in some code to check that the data passed in is _definitely_ an array, because otherwise your code won't work correctly. You might add something like this:
+
+```js
+function uppercaseAll(array) {
+    if (!Array.isArray(array)) {
+        throw new OperationalError({
+            message: 'uppercaseAll function expects an array'
+        });
+    }
+    return array.map(item => item.toUpperCase());
+}
+```
+
+In this case `OperationalError` is **not** the correct error type to use. This is because you, as the publisher of this library, do not know whether a consuming application actually knows how to recover from this error. You can't assume that an error like this is operational – it could easily be a programmer error.
+
+In the above case, you should use JavaScript's built-in errors so that the engineers who maintain the consuming application can catch them and decide for themselves whether they're operational or not:
+
+```js
+function uppercaseAll(array) {
+    if (!Array.isArray(array)) {
+        throw new TypeError('uppercaseAll function expects an array');
+    }
+    return array.map(item => item.toUpperCase());
+}
+```
+
+
+## What does a good error look like?
+
+Good errors are descriptive and can be easily understood by both humans (e.g an engineer trying to debug a problem) and machines (e.g. a logging dashboard used to group errors by type). In this section we'll talk through how to do this.
+
+Whenever you're throwing an error, you should be thinking about the different audiences to whom that error matters. Who are they and what do they need?
+
+  * Yourself in 6 month times when this error starts being thrown at 4am
+
+  * Your logging dashboard, which groups errors so you can easily see when specific ones spike
+
+  * Your first-line support team, who want to know why a system is crashing
+
+### Making errors human readable
+
+All errors in JavaScript support a `message` property, which is a medium-length description of the error that has occurred. `message` is ideal for catering for the "human readable" requirement of an error object, and should be written in a way that humans can read.
```

Comment:
Should there be guidance for coming up with good code names? And perhaps a list of common names? Node.js begins all its errors with `ERR_` which helps me know the source. 

Prefix's and conventions help understand errors when seen in logs so you don't need to always refer to the source code.

Also, should they be unique?

Reply:
I've put in a small note on error code conventions. I think the answer is that I don't know whether we should define conventions yet. I think it might be better if solid conventions come out of the teams who are implementing errors. It's fine for codes to change over time as part of a review that happens later 🙂